src/transformers/models/swin/modeling_swin.py

```diff
@@ -0,0 +1,915 @@
+# coding=utf-8
+# Copyright 2021 temp-swin-author The HuggingFace Inc. team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+""" PyTorch Swin model. """
+
+
+
+
+import collections.abc
+import math
+import os
+
+import torch
+import torch.utils.checkpoint
+from packaging import version
+from torch import nn
+from torch.nn import BCEWithLogitsLoss, CrossEntropyLoss, MSELoss
+
+from ...activations import ACT2FN
+from ...file_utils import (
+    add_code_sample_docstrings,
+    add_start_docstrings,
+    add_start_docstrings_to_model_forward,
+    replace_return_docstrings,
+)
+from ...modeling_outputs import (
+    BaseModelOutput,
+    SequenceClassifierOutput,
+)
+from ...modeling_utils import (
+    PreTrainedModel,
+    apply_chunking_to_forward,
+    find_pruneable_heads_and_indices,
+    prune_linear_layer,
+)
+from ...utils import logging
+from .configuration_swin import SwinConfig
+
+
+logger = logging.get_logger(__name__)
+
+_CHECKPOINT_FOR_DOC = "swin-base"
+_CONFIG_FOR_DOC = "SwinConfig"
+_TOKENIZER_FOR_DOC = "SwinTokenizer"
+
+SWIN_PRETRAINED_MODEL_ARCHIVE_LIST = [
+    "swin-base",
+    # See all Swin models at https://huggingface.co/models?filter=swin
+]
+
+# to_2tuple, drop_path, PatchEmbeddings, PatchMerging and DropPath are from the timm library.
+
+def to_2tuple(x):
+    if isinstance(x, collections.abc.Iterable):
+        return x
+    return (x, x)
```

Comment:
```suggestion
# Copied from transformers.models.vit.modeling_vit.to_2tuple
def to_2tuple(x):
    if isinstance(x, collections.abc.Iterable):
        return x
    return (x, x)
```
At some point, we'll move the to_2tuple function to `image_utils.py`, as it's used by several vision models.